tests/unit/test_gcp_clients.py

```diff
@@ -0,0 +1,250 @@
+        monkeypatch.setattr(class_attribute, fake_url)
+        return fake_url
+
+    @pytest.fixture
+    def compute_rsp(self):
+        return {
+            'kind': 'compute#instanceAggregatedList',
+            'id': 'd34dbeef',
+            # Simplified Instance resource
+            'items': {
+                'us-west1-z': {
+                    'instances': [{
+                        'id': '1',
+                        'creationTimestamp': '2018-01-01 00:00:00.0000',
+                        'name': 'instance-1',
+                        'description': 'guc3-instance-1-54kj',
+                        'tags': {
+                            'items': ['some-tag'],
+                            'fingerprint': ''
+                        },
+                        'machineType': 'n1-standard-1',
+                        'status': 'RUNNING',
+                        'statusMessage': 'RUNNING',
+                        'zone': 'us-west9-z',
+                        'canIpForward': False,
+                        'networkInterfaces': [{
+                            'network': 'network/url/string',
+                            'subnetwork': 'subnetwork/url/string',
+                            'networkIP': '192.168.0.1',
+                            'name': 'test-network',
+                            'accessConfigs': [{
+                                'type': 'ONE_TO_ONE_NAT',
+                                'name': 'EXTERNAL NAT',
+                                'natIP': '1.1.1.1',
+                                'kind': 'compute#accessConfig'
+                            }],
+                        }],
+                        'metadata': {
+                            'items': [
+                                {'key': 'default', 'value': 'true'}
+                            ],
+                        },
+                    }],
+                    'warning': {
+                        'warning': 'object'
+                    }
+                }
+            },
+        }
+
+    @pytest.mark.parametrize('query_str,instance_meta,log_call_count', [
+        # test one page of results + zone filter
+        ('maxResults=10&filter=zone eq us-west9-z', None, 0),
+        # test instance filtering when filtered by tags
+        ('maxResults=10&filter=', ['bl-tag'], 1),
+        # test instance filtering when filtered by metadata
+        ('maxResults=10&filter=', {'key': 'type', 'value': 'c0ffee-b33f'}, 1),
+    ])
+    @pytest.mark.asyncio
+    async def test_list_instances(
+            self, compute_rsp, patch_compute_base_url, get_gce_client, caplog,
+            query_str, instance_meta, log_call_count):
+        """Client uses multiple filters to process results."""
+        caplog.set_level(logging.INFO)
+        gce_client = get_gce_client(gcp_clients.GCEClient)
+
+        if instance_meta:
+            blacklisted_instance = \
+                copy.deepcopy(
```

Comment:
You know, come to think of it, you might not need this if the fixture you're copying is scoped to function (e.g. `@pytest.fixture(scope="function")`, since it'd return a new value every time it's passed into a test method/function.

Reply:
True, however I'm using deepcopy in order to create a second fixture, but with different attributes to test out if blacklisting works.